tape/src/test/java/com/squareup/tape/QueueFileTest.java

```diff
@@ -676,6 +678,26 @@ public void testForEachReadWithOffset() throws IOException {
     assertThat(iteration.get()).isEqualTo(1);
   }
 
+  @Test public void testReadHeaderFailsOnElementTooLarge() throws IOException {
+    File file = copyTestFile(ONE_ENTRY_SERIALIZED_QUEUE);
+    try {
+      QueueFile queueFile = new QueueFile(file, 10);
+      fail("Should have failed with IOException for element too large.");
+    } catch (IOException e) {
+      String message = e.getMessage();
+      assertThat(message).contains("Possible corruption");
+    }
+  }
+
+  @Test public void testElementSizeCorruptionCheckNoProblem() throws IOException {
+    QueueFile queueFile = new QueueFile(file);
+    try {
+      queueFile.peek();
+    } catch (IOException e) {
+      fail("Peek found corruption.", e);
```

Comment:
You can remove the try/catch/fail. If an exception is thrown it will fail the test.